bunch of random things in web worker blog (#100)
NLESC-JCER/run-cpp-on-web: web-worker/README.md

The way the page communicates with the worker object is through sending messages. The page will send a message to the
worker to start doing work, and the message will include all data that the worker needs. The worker then starts
executing the task, using only the data that was in the message. When finished, the worker needs to communicate the
results back to the web app. It will do this by sending a message, so the web app knows when to update.

## The resulting page

> The snippet below shows the web page that uses a web worker. Notice the creation of the `Worker` object, posting of the

Reviewer: ```suggestion
The code snippet below shows the web page that uses a web worker. Notice the creation of the `Worker` object, posting of the
```

</html>
```

The web worker code only contains handling of the incoming message. The web worker unpacks the

> message, does the rootfinding calculation, and packs the results in a new message that it will send back.

Reviewer: ```suggestion
message, does the root finding calculation, and packs the results in a new message that it will send back.
```

The figure below illustrates what is happening in parallel in the two threads.

![threads](threads.png)

We can see the code in action
[here](https://nlesc-jcer.github.io/run-cpp-on-web/web-worker/example-web-worker.html). The calculation still takes
the same time to perform, but as you will notice, the slider remains responsive.

![responsive ui](web-worker.gif)

> _Responsive UI thanks to offloading the rootfinding procedure to the web worker._

Reviewer: ```suggestion
_Responsive UI thanks to offloading the root finding procedure to the web worker._
```

## Recap and what next?

> In this blog post we learned how to keep a web app from freezing while executing computationally intensive C++

Reviewer: ```suggestion
In this blog post, we learned how to keep a web app from freezing while executing computationally intensive C++
```

code. We learned how to create a web worker and how to use the web worker in a simple web app. 

> Now take your web app a step further by reading some of the other blogs in this series, for example [how to interact with your app using forms](../react/README.md) and [how to spice up your app up with

Reviewer: ```suggestion
Now you can take this web app a step further by reading some of the other blogs in this series, for example [how to interact with your app using forms](../react/README.md) and [how to spice up your app up with
```

visualizations](../vega/README.md). A final blog shows you [how to combine all of the above in a single web app](../kitchen-sink/README.md).

Looking for the first blog and introduction to this series? [This blog post](../webassembly/README.md) will show you how
to get started running your C++ on the web using webassembly.

## Get in touch with us

> This blog was written by the Netherlands eScience Center's Generalization Team. The team consists of Stefan Verhoeven,

Reviewer: ```suggestion
This blog was written by Generalization Team of Netherlands eScience Center.  The team consists of Stefan Verhoeven,
```
